### PR TITLE
Fix RS0033 and RS0034 in the presence of primary constructors

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Depends on https://github.com/dotnet/roslyn-analyzers/pull/6690 going in to update hte version of Roslyn we support.

### CODE REVIEW

File: src/Roslyn.Diagnostics.Analyzers/UnitTests/ExportedPartsShouldHaveImportingConstructorTests.cs [hunk 1]
```diff
@@ -39,6 +39,35 @@ class C {{
...
+        [InlineData("System.ComponentModel.Composition")]
+        public async Task SingleExpectedConstructor_PrimaryConstructors_CSharpAsync(string mefNamespace)
+        {
+            var source = $$"""
+                using {{mefNamespace}};
+                using System.Diagnostics.CodeAnalysis;
+
+                [Export]
+                [method: SuppressMessage("RoslynDiagnosticsReliability", "RS0034:Exported parts should have [ImportingConstructor]", Justification = "Used incorrectly by tests")]
+                class C(string s)
```
Comment (CyrusNajmabadi) [inline src/Roslyn.Diagnostics.Analyzers/UnitTests/ExportedPartsShouldHaveImportingConstructorTests.cs:53]: @sharwell @mavasani The testign harness doesn't parse this properly (maybe it's using an older version of roslyn?). Could you let me know what i need to update to get primary constructors understood by roslyn-analyzers tests?

File: src/Roslyn.Diagnostics.Analyzers/UnitTests/ExportedPartsShouldHaveImportingConstructorTests.cs [hunk 2]
```diff
@@ -39,6 +39,35 @@ class C {{
...
+        [InlineData("System.ComponentModel.Composition")]
+        public async Task SingleExpectedConstructor_PrimaryConstructors_CSharpAsync(string mefNamespace)
+        {
+            var source = $$"""
+using {{mefNamespace}};
+using System.Diagnostics.CodeAnalysis;
+
+[Export]
+[method: SuppressMessage("RoslynDiagnosticsReliability", "RS0034:Exported parts should have [ImportingConstructor]", Justification = "Used incorrectly by tests")]
+class C(string s)
```
[the remark this answers is not in the document]
Comment (sharwell) [reply src/Roslyn.Diagnostics.Analyzers/UnitTests/ExportedPartsShouldHaveImportingConstructorTests.cs]: ➡️ This is now resolved

File: src/Roslyn.Diagnostics.Analyzers/UnitTests/ExportedPartsShouldHaveImportingConstructorTests.cs (hunk 1 above)
Comment (sharwell) [reply src/Roslyn.Diagnostics.Analyzers/UnitTests/ExportedPartsShouldHaveImportingConstructorTests.cs:53]: ➡️ This is now resolved

**Comment by sharwell** [comment]
@CyrusNajmabadi test failures may be analyzer bugs

**Pull request closed without merging**